providers/aws-ec2/src/main/java/org/jclouds/aws/ec2/features/RouteTableApi.java

```diff
@@ -0,0 +1,277 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.aws.ec2.features;
+
+import static org.jclouds.aws.reference.FormParameters.ACTION;
+
+import javax.inject.Named;
+import javax.ws.rs.FormParam;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+
+import org.jclouds.Fallbacks;
+import org.jclouds.aws.ec2.binders.BindRouteTableIdsToIndexedFormParams;
+import org.jclouds.aws.ec2.domain.RouteTable;
+import org.jclouds.aws.ec2.options.RouteTableOptions;
+import org.jclouds.aws.ec2.xml.AssociateRouteTableResponseHandler;
+import org.jclouds.aws.ec2.xml.CreateRouteTableResponseHandler;
+import org.jclouds.aws.ec2.xml.DescribeRouteTablesResponseHandler;
+import org.jclouds.aws.ec2.xml.ReturnValueHandler;
+import org.jclouds.aws.filters.FormSigner;
+import org.jclouds.javax.annotation.Nullable;
+import org.jclouds.location.functions.RegionToEndpointOrProviderIfNull;
+import org.jclouds.rest.annotations.BinderParam;
+import org.jclouds.rest.annotations.EndpointParam;
+import org.jclouds.rest.annotations.Fallback;
+import org.jclouds.rest.annotations.FormParams;
+import org.jclouds.rest.annotations.RequestFilters;
+import org.jclouds.rest.annotations.VirtualHost;
+import org.jclouds.rest.annotations.XMLResponseParser;
+
+import com.google.common.collect.FluentIterable;
+
+/**
+ * Provides access to AWS Route Table services.
+ *
+ * @see <a href="http://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_RouteTable.html">RouteTable docs</a>
+ */
+@RequestFilters(FormSigner.class)
+@VirtualHost
+@Path("/")
+public interface RouteTableApi {
+
+   /**
+    * Creates a {@link RouteTable}
+    *
+    * @param region The region to create the table in.
+    * @param vpcId The ID of the VPC
+    * @return The route table
+    */
+   @Named("CreateRouteTable")
+   @POST
+   @FormParams(keys = ACTION, values = "CreateRouteTable")
+   @XMLResponseParser(CreateRouteTableResponseHandler.class)
+   RouteTable createRouteTable(
+      @EndpointParam(parser = RegionToEndpointOrProviderIfNull.class) @Nullable String region,
+      @FormParam("VpcId") String vpcId);
+
+   /**
+    * Creates a {@link RouteTable}, supplying options.
+    *
+    * @param region  The region to create the table in
+    * @param vpcId The ID of the VPC
+    * @param options Options for the request
+    * @return The route table
+    */
+   @Named("CreateRouteTable")
+   @POST
+   @FormParams(keys = ACTION, values = "CreateRouteTable")
+   @XMLResponseParser(CreateRouteTableResponseHandler.class)
+   RouteTable createRouteTable(
+      @EndpointParam(parser = RegionToEndpointOrProviderIfNull.class) @Nullable String region,
+      @FormParam("VpcId") String vpcId,
+      RouteTableOptions options);
```

Comment:
This request actually only supports the dryRun parameter. Worth changing the signature to just accept a boolean.
It looks like the options object is more for the `Route` than the `RouteTable` objects. Worth renaming to `RouteOptions`, and create specific options objects for those methods that require them? Otherwise, it can be confusing for users which options can be used with each method. Let's expose just what can be used.